src/parseMessageTest.c

```diff
@@ -0,0 +1,134 @@
+#ifdef UNIT_TEST
+
+#include "parseMessage.c"
+#include <stdio.h>
+#include <assert.h>
+
+void test_parse_u8() {
+   uint8_t message[] = {1, 2};
+   MessageParser parser = {message, sizeof(message)};
+   uint8_t value;
+   assert(parse_u8(&parser, &value) == 0);
+   assert(parser.buffer_length == 1);
+   assert(parser.buffer == message + 1);
+   assert(value == 1);
+}
+
+void test_parse_u8_too_short() {
+   uint8_t message[] = {};
+   MessageParser parser = {message, sizeof(message)};
+   uint8_t value;
+   assert(parse_u8(&parser, &value) == 1);
+}
+
+void test_parse_length() {
+   uint8_t message[] = {1, 2};
+   MessageParser parser = {message, sizeof(message)};
+   size_t value;
+   assert(parse_length(&parser, &value) == 0);
+   assert(parser.buffer_length == 1);
+   assert(parser.buffer == message + 1);
+   assert(value == 1);
+}
+
+void test_parse_length_two_bytes() {
+   uint8_t message[] = {128, 1};
+   MessageParser parser = {message, sizeof(message)};
+   size_t value;
+   assert(parse_length(&parser, &value) == 0);
+   assert(parser.buffer_length == 0);
+   assert(parser.buffer == message + 2);
+   assert(value == 128);
+}
+
+void test_parse_message_header() {
+   uint8_t message[] = {1, 2, 3};
+   MessageParser parser = {message, sizeof(message)};
+   MessageHeader header;
+   assert(parse_message_header(&parser, &header) == 0);
+   assert(parser.buffer_length == 0);
+   assert(parser.buffer == message + 3);
+}
+
+void test_parse_pubkeys() {
+   uint8_t message[33] = {1};
+   MessageParser parser = {message, sizeof(message)};
+   Pubkey* pubkeys;
+   size_t pubkeys_length;
+   assert(parse_pubkeys(&parser, &pubkeys, &pubkeys_length) == 0);
+   assert(parser.buffer_length == 0);
+   assert(parser.buffer == message + 33);
+}
+
+void test_parse_pubkeys_too_short() {
+   uint8_t message[] = {1};
+   MessageParser parser = {message, sizeof(message)};
+   Pubkey* pubkeys;
+   size_t pubkeys_length;
+   assert(parse_pubkeys(&parser, &pubkeys, &pubkeys_length) == 1);
+}
+
+void test_parse_blockhash() {
+   uint8_t message[32];
+   MessageParser parser = {message, sizeof(message)};
+   Blockhash* blockhash;
+   assert(parse_blockhash(&parser, &blockhash) == 0);
+   assert(parser.buffer_length == 0);
+   assert(parser.buffer == message + 32);
+}
```

Comment:
Test that the blockhash contains the correct data?

Reply:
Fixed